src/wp-includes/block-template-utils.php

```diff
@@ -229,14 +229,12 @@ function _get_block_templates_paths( $base_directory ) {
 		return $template_path_list[ $base_directory ];
 	}
 	$path_list = array();
-	try {
+	if ( is_dir( $base_directory ) ) {
```

Comment:
Instead of checking here could we add condition at top?

```
if ( ! is_dir( $base_directory ) ) {
```

Reply:
I think we still want to memoize the path list even when there is no directory, otherwise the `is_dir()` check would get repeated unnecessarily, so keeping it as is should be fine.